x-pack/libbeat/management/enroll.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package management
+
+import (
+	"github.com/elastic/beats/libbeat/cmd/instance"
+	"github.com/elastic/beats/x-pack/libbeat/management/api"
+)
+
+// Enroll this beat to the given kibana
+// This will use Central Management API to enroll and retrieve an access key for config retrieval
+func Enroll(beat *instance.Beat, kibanaURL, enrollmentToken string) error {
+	config, err := api.ConfigFromURL(kibanaURL)
+	if err != nil {
+		return err
+	}
+
+	client, err := api.NewClient(config)
+	if err != nil {
+		return err
+	}
+
+	accessToken, err := client.Enroll(beat.Info.Beat, beat.Info.Version, beat.Info.Hostname, beat.Info.UUID, enrollmentToken)
+	if err != nil {
+		return err
+	}
+
+	// Enrolled, persist state
+	// TODO use beat.Keystore() for access_token
```

Comment:
I think we would keep management.yml, as it contains structured data and state, keystore doesn't seem the best place for this as it may be too obscure?

Reply:
I would consider this file similar to Filebeat registry, it's something internal, but good to have it around for debugging / understanding what's going on

Reply:
Yes, agree we should keep it around. I wonder if we should in the future use the new "registry" writer for it, but still a different file. So all our state documents have the same structure (@urso FYI).